sigstore-java/src/main/java/dev/sigstore/tuf/Updater.java

```diff
@@ -221,13 +225,26 @@ void verifyDelegate(
           } else {
             pubKey = Keys.constructTufPublicKey(Hex.decode(publicKeyContents), key.getScheme());
           }
-          byte[] signatureBytes = Hex.decode(signature.getSignature());
           try {
+            // while we error on keys that are not readable, we are intentionally more permissive
+            // about signatures. If for ANY reason (except unparsed keys) we cannot validate a
+            // signature, we continue as long as we find enough valid signatures within the
+            // threshold
+            byte[] signatureBytes = Hex.decode(signature.getSignature());
             if (verifiers.newVerifier(pubKey).verify(verificationMaterial, signatureBytes)) {
               goodSigs.add(signature.getKeyId());
             }
-          } catch (SignatureException e) {
-            throw new TufException(e);
+          } catch (SignatureException
+              | DecoderException
+              | NoSuchAlgorithmException
+              | InvalidKeyException e) {
+            log.log(
+                Level.FINE,
+                "TUF: ignored invalid signature: "
+                    + signature.getSignature()
```

Comment:
I am afraid it would be prone to log injection. The message will be passed through messageformat anyway. Wdyt?

Reply:
Yes

Reply:
alright cool lemme fix that

Reply:
Should we also just switch to using a slf4j or log4j?

Reply:
I would be -1..-0.5 re log4j as their backward compatibility is far from being solid (e.g. they completely ignore CVEs in log4j1, and they might be breaking backward compatibility betwen log4j2-api and log4j3-api)
It might be ok to use slf4j

Reply:
yeah, well I guess we can consider that. We don't do much logging anyway, so maybe jul is fine for now.